frontends/PyCDE/src/pycde/system.py

```diff
@@ -59,6 +61,7 @@ def run_passes(self):
       return
     pm = mlir.passmanager.PassManager.parse(",".join(self.passes))
     pm.run(self.mod)
+    types.declare_types(self.mod)
```

Comment:
@teqdruid curious what you think about this... I was wondering where to put this. I think this makes sense. In terms of phase ordering, there may be potentially several calls to `generate` before a module is fully generated. This is, to me, a "pass" written in Python, which falls under the category of "things we have to do to legalize the IR before ExportVerilog".